Components/Core/Timer.cpp

```diff
@@ -21,14 +21,63 @@ Timer::Timer()
 	// The timer ID is specified as (void *)this to provide a unique ID for each timer object - however this is not necessary for polling timers.
 	// The timer is created in the dormant state.
 	rtTimerHandle = xTimerCreate("Timer", DEFAULT_TIMER_PERIOD, pdFALSE, (void *)this, empty_callback);
+	SOAR_ASSERT(rtTimerHandle, "Error Occurred, Timer not created");
+	timerState = UNINITIALIZED;
 }
 
+
+
 /**
  * @brief Changes this timer object's RTOS timer period, returns true on success, returns false on failure (timer command queue full)
  */
-bool Timer::ChangePeriod(const uint32_t period)
+bool Timer::ChangePeriod(const uint32_t period_ms)
+{
+	if (xTimerChangePeriod(rtTimerHandle, MS_TO_TICKS(period_ms), DEFAULT_TIMER_COMMAND_WAIT_PERIOD) == pdTRUE) {
+		StopTimer();
+		timerState = PAUSED;
+		return true;
+	}
+	return false;
+}
+
+/**
+ * @brief Starts the timer, returns true on success, returns false on failure
+*/
+bool Timer::StartTimer()
+{
+	if ((timerState == COMPLETE) || (timerState == COUNTING)) {
```

Comment:
This is purely coding style, but I would make this a flat structure.

Eg. make the first statement on it's own and make the else if an if statement (since the first statement returns anyway)

Reply:
Done